### PR TITLE
Introducing: Users service

### PR DESCRIPTION
**Pull request by crufter**
Picked up Asim's SQL users service (see PR #16) and ported it to store.
Store still has room for improvements which I will address later.

I also plan to add tests for the users service soon.

### CODE REVIEW

File: users/db/db.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	"errors"
+	"time"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/micro/dev/model"
+	user "github.com/micro/services/users/proto"
+)
+
+var (
+	Accounts model.DB
```
Comment (crufter) [inline users/db/db.go:13]: These are only globals so the first commit is a minimal delta from the existing users service code.

File: users/db/db.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	if len(sess) == 0 {
+		return nil, errors.New("Not found")
+	}
+	return sess[0], nil
+}
+
+func Create(user *user.User, salt string, password string) error {
+	user.Created = time.Now().Unix()
+	user.Updated = time.Now().Unix()
+	// @todo what to do with salt and password?
```
Comment (crufter) [inline users/db/db.go:49]: I think salt and password should be it's own table?

File: users/db/db.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+func Update(user *user.User) error {
+	user.Updated = time.Now().Unix()
+	return Accounts.Save(user)
+}
+
+func Read(id string) (*user.User, error) {
+	users := []*user.User{}
+	// @todo there should be a Read in the model to get rid of this pattern
```
Comment (crufter) [inline users/db/db.go:64]: I think this pattern clearly shows the need for a `Read` method on the `model.DB`?
It comes up pretty frequently.

Comment (asim) [reply users/db/db.go:64]: Yes model should implement CRUD.

File: users/db/db.go (hunk 2 above)
Comment (asim) [reply users/db/db.go:49]: Likely if you think it makes sense. As per our discussion, future optimisation is the redaction of sensitive fields.

**Review by asim — APPROVED** [review]

**Pull request merged**